core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -442,7 +452,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       numPendingExecutors += knownExecutors.size
     }
 
-    doKillExecutors(executorsToKill)
+    (force || !executorsToKill.isEmpty) && doKillExecutors(executorsToKill)
```

Comment:
Just `doKillExecutors` wouldn't work; this needs to return `false` when executors are not killed, otherwise the code to cleanup `executorsPendingToRemove` in `ExecutorAllocationManager` would be needed.

But the return can be simplified:

    executorsToKill.nonEmpty && doKillExecutors(executorsToKill)

The list can only be empty when `force` is `false` (or an empty list was provided in the first place), so checking `force` is redundant.